### PR TITLE
update netlify github action to 3.0

### PR DESCRIPTION
**Pull request by tsloughter**

### CODE REVIEW

**Comment by tsloughter** [comment]
This hopefully fixes the CI failure issue once merged to main.

**Review by ferd — APPROVED** [review]

**Comment by ferd** [comment]
I think we had issues with this in other repos in the past: https://github.com/erlang/rebar3/commit/a28d57ef25347d0cce25e93680ce75ee6c0dcf8a

Basically the github permissions changed and to write a comment about the deployment back in this issue, you must give the workflow explicit write permissions to the repo.

**Comment by tsloughter** [comment]
Hm, so this has the write permission but its within the `job` while in `rebar3` it is at the top level. I'll move this one to the top level to be sure.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,13 +8,14 @@ on:
       - "snapshot-*"
   pull_request:
 
+permissions:
+  contents: write
```
Comment (ferd) [inline .github/workflows/ci.yml:12]: `contents` was used for rebar3 releases. For this one you need it to be `issues` to create comments.

**Pull request merged**